cooperator/models/res_partner.py

```diff
@@ -283,3 +285,21 @@ def create_cooperative_membership(self, company_id):
                 }
             )
         return result
+
+    def get_share_quantities(self):
+        """Return a defaultdict(int) with the amount of shares per product id."""
+        self.ensure_one()
+        total_shares = defaultdict(int)
+
+        for line in self.share_ids:
+            total_shares[line.share_product_id.id] = line.share_number
+
+        return total_shares
+
+    def get_total_shares(self):
+        """Return the total amount of shares belonging to a partner."""
+        self.ensure_one()
+        amount = 0
+        for qty in self.get_share_quantities().values():
+            amount += qty
+        return amount
```

Comment:
no need for this function: the `res.partner.number_of_share` (actually, `cooperative.membership.number_of_share`) field already contains this information.